LoopStructural/modelling/core/geological_model.py

```diff
@@ -1046,7 +1054,157 @@ def create_and_add_folded_fold_frame(
-        return folded_fold_frame
+        return fold_frame
+
+    def create_and_add_intrusion(
+        self,
+        intrusion_name,
+        intrusion_frame_name=None,
+        intrusion_network_type=None,
+        intrusion_network_contact=None,
+        contacts_anisotropies=None,
+        structures_anisotropies=None,
+        sequence_anisotropies=None,
+        inet_axis=None,
+        intrusion_lateral_extent_model=None,
+        intrusion_vertical_extent_model=None,
+        lateral_extent_sgs_parameters={},
+        vertical_extent_sgs_parameters={},
+        **kwargs,
+    ):
+        """
+
+        Parameters
+        ----------
+        intrusion_name :  string, 
+            name of intrusion feature in model data
+        intrusion_frame_name :  string, 
+            name of intrusion frame in model data
+        intrusion_network_type :  string, 
+            algorithm to build intrusion network 'interpolated' or 'shortest path'
+        intrusion_network_contact :  string, 
+            name of contact (roof or floor) to be used to build intrusion network
+        contacts_anisotropies : list, 
+            name of stratigraphic units where intrusion is emplaced
+        structures_anisotropies : list, 
+            name of structures exploited by intrusion
+        sequence_anisotropies = list, 
+            name of anisotropies to look for the shortest path. It could be only starting and end point.
+        intrusion_lateral_extent_model = function, 
+            geometrical conceptual model for simulation of lateral extent
+        intrusion_vertical_extent_model = function, 
+            geometrical conceptual model for simulation of vertical extent
+        lateral_extent_sgs_parameters = dictionary, 
+            parameters for sequential gaussian simulation of lateral extent
+        vertical_extent_sgs_parameters = dictionary, 
+            parameters for sequential gaussian simulation of vertical extent
+
+        kwargs
+
+        Returns
+        -------
+        intrusion feature
+
+        """
+        if intrusions == False:
+            logger.error("Libraries not installed")
+            raise Exception("Libraries not installed")
+        feature_data = self.data[self.data["feature_name"] == intrusion_name].copy()
+
+        # Create and build Intrusion Network
+        INet = IntrusionNetwork(
+            feature_data=feature_data,
+            intrusion_network_contact=intrusion_network_contact,
+            intrusion_network_type=intrusion_network_type,
+            model=self,
+            **kwargs,
+        )
+
+        INet.set_data()
+        INet.set_contact_anisotropies(contacts_anisotropies, **kwargs)
+        INet.set_faults_anisotropies(structures_anisotropies)
+        INet.set_sequence_of_exploited_anisotropies(sequence_anisotropies)
+        INet.set_velocity_parameters()
+        INet.set_sections_axis(inet_axis)
+        logger.info("building intrusion network")
+        INet.build(**kwargs)
```

Comment:
Can we avoid running "build" within the create_and_add function? I have tried to move all of the interpolation to be run on demand when the geological feature is evaluated or when builder.update is called